File: examples/bimapi/CheckbotBimLink/FeaExample/BimApiLinkFeaExample/Importers/LoadGroupImporter.cs
using BimApiLinkFeaExample.FeaExampleApi;
using IdeaStatiCa.BimApi;
using IdeaStatiCa.BimApiLink.BimApi;
using IdeaStatiCa.BimApiLink.Importers;

namespace BimApiLinkFeaExample.Importers
{
	internal class LoadGroupImporter : IntIdentifierImporter<IIdeaLoadGroup>
	{
		private readonly IFeaLoadsApi loadsApi;

		public LoadGroupImporter(IFeaLoadsApi loadsApi)
		{
			this.loadsApi = loadsApi;
		}

		public override IIdeaLoadGroup Create(int id)
		{
			IFeaLoadGroup loadGroup = loadsApi.GetLoadGroup(id);
			return new IdeaLoadGroup(id)
			{
				Name = loadGroup.Name,
				GroupType = GetLoadGroupType(loadGroup.LoadGroupCategory)
			};
		}

		private static IdeaRS.OpenModel.Loading.LoadGroupType GetLoadGroupType(LoadGroupCategory loadGroupCategory)
		{ 
			switch(loadGroupCategory) 
			{
				case LoadGroupCategory.Permanent:
					return IdeaRS.OpenModel.Loading.LoadGroupType.Permanent;
				case LoadGroupCategory.Variable:
					return IdeaRS.OpenModel.Loading.LoadGroupType.Variable;
				default:
					return IdeaRS.OpenModel.Loading.LoadGroupType.Permanent;
			}
		}
	}
}

File: examples/bimapi/CheckbotBimLink/CppFeaExample/CppFeaApiWrapper/LoadGroupImporter.cpp
#include "pch.h"
#include "LoadGroupImporter.h"
#include "..\CppFeaApi\NativeFeaLoads.h"

namespace CppFeaApiWrapper
{
	namespace Importers
	{
		LoadGroupImporter::LoadGroupImporter(ImporterContext^ context)
		{
			this->context = context;
		}

		IIdeaLoadGroup^ LoadGroupImporter::Create(int id)
		{
			auto loadGroup = this->context->GetApi()->GetLoading()->GetLoadGroup(id);

			IdeaLoadGroup^ ideaLoadGroup = gcnew IdeaLoadGroup(id);
			ideaLoadGroup->Name = gcnew String(loadGroup->Name.c_str());
			ideaLoadGroup->GroupType = GetLoadGroupType(loadGroup->LoadGroupCategory);

			return ideaLoadGroup;
		}

		IdeaRS::OpenModel::Loading::LoadGroupType LoadGroupImporter::GetLoadGroupType(int loadGroupCategory)
		{
			switch (loadGroupCategory) {
			case NativeFeaLoads::LoadGroup_Category_Permanent:
				return IdeaRS::OpenModel::Loading::LoadGroupType::Permanent;
			case NativeFeaLoads::LoadGroup_Category_Variable:
				return IdeaRS::OpenModel::Loading::LoadGroupType::Variable;
			default:
				return IdeaRS::OpenModel::Loading::LoadGroupType::Permanent;
			}
		}
	}
}
